src/tools/schema2kotlin.ts

```diff
@@ -26,85 +22,59 @@ const keywords = [
   'suspend', 'tailrec', 'vararg', 'field', 'it'
 ];
 
-function leftPad(block: string, n: number = 0) {
-  return block.split('\n')
-    .map((line) => `${' '.repeat(n)}${line}`)
-    .join('\n');
-}
-
-function generate(name: string, schema): string {
-  const fields: string[] = [];
-  const encode: string[] = [];
-  const decode: string[] = [];
-
-  const processValue = (field, typeChar): void => {
-    const typeMap = {
-      'T': ['String', '""'],
-      'U': ['String', '""'],
-      'N': ['Double', '0.0'],
-      'B': ['Boolean', 'false'],
-    };
-
-    const decodeMap = {
-      'T': 'decodeText()',
-      'U': 'decodeText()',
-      'N': 'decodeNum()',
-      'B': 'decodeBool()'
-    };
-    const type = typeMap[typeChar][0];
-    const defaultVal = typeMap[typeChar][1];
-    const decodeType = typeMap[typeChar];
-
-    const fixed = field + (keywords.includes(field) ? '_' : '');
-
-    fields.push(`var ${fixed}: ${type} = ${defaultVal}`);
-
-
-    decode.push(`"${fixed}" -> {`,
-      `     decoder.validate("${typeChar}")`,
-      `     ${fixed} = decoder.${decodeType}`,
-      `}`,
-    );
+// type-char to [kotlin-type, default-value, decode-functino]
+const typeMap = {
+  'T': ['String', '""', 'decodeText()'],
+  'U': ['String', '""', 'decodeText()'],
+  'N': ['Double', '0.0', 'decodeNum()'],
+  'B': ['Boolean', 'false', 'decodeBool()'],
+};
+
+export class Schema2Kotlin extends Schema2Base {
+  // test-Kotlin.file_name.manifest -> TestKotlinFileNameManifest.kt
+  outputName(baseName: string): string {
+    const parts = baseName.toLowerCase().replace(/\./g, '-').split(/[-_]/);
+    return parts.map(part => part[0].toUpperCase() + part.slice(1)).join('') + '.kt';
```

Comment:
Nice TitleCase logic.

Reply:
I would also be happy to see unit tests, here.

Reply:
Ditto.